integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonAddLoadCommand.scala

```diff
@@ -92,6 +92,23 @@ case class CarbonAddLoadCommand(
     val segmentPath = options.getOrElse(
       "path", throw new UnsupportedOperationException("PATH is manadatory"))
 
+    val allSegments = SegmentStatusManager.readLoadMetadata(carbonTable.getMetadataPath)
+
+    // If a path is already added then we should block the adding of the same path again.
+    allSegments.exists { a =>
+      a.getPath != null && a.getPath.equalsIgnoreCase(segmentPath)
+      throw new AnalysisException(s"Cannot add the segment. This path is already in use by " +
+                                  s"another segment.")
+    }
+
+    val format = options.getOrElse("format", "carbondata")
+    val isCarbonFormat = format.equalsIgnoreCase("carbondata") || format.equalsIgnoreCase("carbon")
+
+    // If in the given location no carbon files are found then we can throw an exception
+    if (isCarbonFormat && SegmentFileStore.getListOfCarbonIndexFiles(segmentPath).isEmpty) {
```

Comment:
I think this validation should come from SparkFiileFormat, so ideally there should not be any changes in this class for format level validation. Better move it to format level

Reply:
@ravipesala Here we are just checking the presence of carbon files present in the current folder location and as there was a 'isCarbonFormat' below so I just shifted it above and used it.

Reply:
If we need to check this, better to check weather index file and data file are matching:
1. If it has merge index file, all data file should be present in merge index file
2. otherwise, one index file for one data file should present
please move this validation to another function and invoke it here

Reply:
@manishnalla1994  the condition earlier is not for validation, it is just to update the metadata. How is the behavior of SparkFiileFormat infer schema if there are no carbondata or carbonindex files? I feel it is better to correct in the down layer than adding validations on top layer.